ddtrace/_trace/_span_link.py

```diff
@@ -117,3 +124,55 @@ def __str__(self) -> str:
             f"trace_id={self.trace_id} span_id={self.span_id} attributes={attrs_str} "
             f"tracestate={self.tracestate} flags={self.flags} dropped_attributes={self._dropped_attributes}"
         )
+
+
+# Span Pointers are currently private, so let's put them here for now
```

Comment:
as this api gets more interesting and we add actual span linking functionality, we are going to be moving this code out to `ddtrace/_trace/_span_pointers.py` (I have a version of this on another branch now, but getting it ported here would be a bit of a hassle. let me know if anyone feels strongly that this should start in that other file, otherwise we'll leave it to the next PR)

Reply:
here's that next PR: https://github.com/DataDog/dd-trace-py/pull/10639/